lib/es6/caml_obj.js

```diff
@@ -358,40 +386,56 @@ function equal_nullable(x, y) {
   }
 }
 
+function isNumberOrBigInt(a) {
+  if (typeof a === "number") {
+    return true;
+  } else {
+    return typeof a === "bigint";
+  }
+}
+
+function canNumericCompare(a, b) {
+  if (isNumberOrBigInt(a)) {
+    return isNumberOrBigInt(b);
+  } else {
+    return false;
+  }
+}
+
 function notequal(a, b) {
-  if (typeof a === "number" && typeof b === "number") {
+  if (canNumericCompare(a, b)) {
```

Comment:
It's better inlined.
I suspect this will be inlined again once canNumericCompare is made a private function.

Reply:
I'm not sure it is re-building. I changed it to == number or == string or == bigint but the javascript is the same. What command to build with rescript? `make lib` not doing it. I have the reScript extension on my computer and installed Rescript into package.json.

Reply:
Need to go to sleep. Do you think I need tests related to comparing bigInt and float? In JavaScript you can type bigInt < 3.0 but I think ReScript prevents this unless you do an unsafe cast. Will look at this in the morning.

Reply:
I can take it from here. Thanks for this!